expression/builtin_cast.go

```diff
@@ -573,6 +573,15 @@ func (b *builtinCastIntAsTimeSig) evalTime(row chunk.Row) (res types.Time, isNul
 	if isNull || err != nil {
 		return res, isNull, err
 	}
+
+	if b.args[0].GetType().Tp == mysql.TypeYear {
```

Comment:
Please also update `builtinCastIntAsTimeSig.vecEvalTime`.~

Reply:
ok, I will do this late today.